src/components/Card.js

```diff
@@ -1,55 +1,88 @@
 export default class Card {
-    constructor(data, cardSelector, handleImageClick) {
-        this.name = data.name;
-        this.link = data.link;
-        this._cardSelector = cardSelector;
-        this._handleImageClick = handleImageClick;
+    constructor(
+      data,
+      cardSelector,
+      handleImageClick,
+      handleDeleteCard,
+      handleLikeToggle,
+      userId
+    ) {
+      this.name = data.name;
+      this.link = data.link;
+      this._id = data._id;
+      this._isLiked = data.isLiked;
+      this._userId = userId;
+      this._cardSelector = cardSelector;
+      this._handleImageClick = handleImageClick;
+      this._handleDeleteCard = handleDeleteCard;
+      this._handleLikeToggle = handleLikeToggle;
     }
-
+  
+    _isLikedByUser() {
+      return this._isLiked.some((like) => like._id === this._userId);
+    }
+  
+    _updateLikeStatus() {
+      const likeButton = this._cardElement.querySelector(".card__heart");
```

Comment:
**Could be improved**

All elements found **every time** you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `getView`) only **1 time** in order not to waste resources on searching them again and again.

`Class fields` (the so-called **class variables** with `this`) are accessible inside any class method. You shouldn’t pass **class fields** into the call of any method or search them again because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form`, `this._likeButton`, and so on.

```js
  getView() {
    this._element = this._getTemplate();
    this._likeButton = this._element.querySelector(".card__like-button");
    this._cardImage = this._element.querySelector(".card__image");
    this._trashButton = this._element.querySelector(".card__trash-button");

```